samples/sample_common/src/vaapi_utils.cpp

```diff
@@ -451,7 +451,9 @@ void CLibVA::ReleaseVASurface(
         AcquireCtx* ctx = (AcquireCtx*)actx;
         if (ctx) {
             m_libva.vaDestroySurfaces(dpy2, &srf2, 1);
-            close(ctx->fd);
+            if (ctx->fd) {
```

Comment:
This is wrong check. You must check for `>=0` since that's a range of valid fds:
```
if (ctx->fd >= 0) {
    close(ctx->fd);
}
```

Reply:
the current issue is that default ctx->fd value is 0, but it is a invalid fd, if we closed it here, it has chance to close the fd 0 which is the file opened by other component and cause trouble to overall appliction.

Reply:
> the current issue is that default ctx->fd value is 0

Where it was set to be 0? Can we change the default to -1 which is more appropriate?

Reply:
@dvrogozh  I updated the code based on your feedback.